### PR TITLE
Make Tab an unsealed runtimeclass (and rename it to TabBase)

### LINKED ISSUE #7969: Close Menu, "Close other tabs" and "Close tabs to right" always operate on tab index 0, are always enabled
I'm not sure if this is an issue in main or an issue in feature/settings-ui.

/cc @mpela81

### PR DESCRIPTION
**Pull request by DHowett**
In preparation for the Settings UI, we needed to make some changes to
Tab to abstract out shared functionality common between different types
of tab. This is the result of that work. All code references to the
settings have been removed or reverted.

Contains changes from #8053, #7802.

The messages below only make sense in the context of the Settings UI,
which this pull request does not bring in. They do, however, provide
valuable information.

From #7802 (@leonMSFT):

> This PR's goal was to add an option to the `OpenSettings` keybinding to
> open the Settings UI in a tab. In order to implement that, a couple of
> changes had to be made to `Tab`, specifically:
>
> - Introduce a tab interface named `ITab`
> - Create/Rename two new Tab classes that implement `ITab` called
>   `SettingsTab` and `TerminalTab`
>

From #8053:

> `TerminalTab` and `SettingsTab` share some implementation details. The
> close submenu introduced in #7728 is a good example of functionality
> that is consistent across all tabs. This PR transforms `ITab` from an
> interface, into an [unsealed runtime class] to de-duplicate some
> functionality. Most of the logic from `SettingsTab` was moved there
> because I expect the default behavior of a tab to resemble the
> `SettingsTab` over a `TerminalTab`.
>
> ## References
> Verified that Close submenu work was transferred over (#7728, #7961, #8010).
>
> ## Validation Steps Performed
> Check close submenu on first/last tab when multiple tabs are open.
>
> Closes #7969
>
> [unsealed runtime class]: https://docs.microsoft.com/en-us/uwp/midl-3/intro#base-classes

Co-authored-by: Carlos Zamora <carlos.zamora@microsoft.com>

### CODE REVIEW

**Comment by DHowett** [comment]
I was _one of the two signoffs_ on both of these pull requests, so I will merge this if @zadjii-msft, @miniksa or @PankajBhojwani sign off. Highly preferable that it's @zadjii-msft.

@carlos-zamora and @leonMSFT do not count.

File: src/cascadia/TerminalApp/TerminalPage.cpp
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1559]: There's a lot of this `if (get focused tab index) { if (get tab from index) { tab.doSomething() } }` going on in here. Would it be better to have a helper that just does it all in one go?

File: src/cascadia/TerminalApp/Tab.idl
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/Tab.idl:10]: So just as a sanity check, `IconSource` wasn't actually being used anywhere, was it?

**Review by zadjii-msft — APPROVED** [review]
I'm okay with this.

I think in the future, it'd be better to have it all be a uniform `Tab` with non-terminal _panes_, but we can work towards that from this. SUI is way higher priority now anyways.

File: src/cascadia/TerminalApp/TerminalPage.cpp
Comment (DHowett) [reply src/cascadia/TerminalApp/TerminalPage.cpp:1559]: I’d rather not add that now to reduce the merge burden when this flows _back_ into SUI (since right now i can use the “take all `ours`” strategy), but it is worth doing.

File: src/cascadia/TerminalApp/Tab.idl
Comment (DHowett) [reply src/cascadia/TerminalApp/Tab.idl:10]: On account of it compiles and runs fine without IconSource, I am inclined to say yes.

**Comment by DHowett** [comment]
@msftbot merge this in about 1 minute

**Comment by DHowett** [comment]
Oh right, bot don't merge single signoff PRs

**Comment by miniksa** [comment]
lol I was just about to 2nd this.

**Pull request merged**
